src/test/java/teammates/test/pageobjects/InstructorFeedbackResultsPage.java

```diff
@@ -332,6 +332,58 @@ public void hoverAndViewStudentPhotoOnBody(int panelBodyIndex, String urlRegex)
         JavascriptExecutor jsExecutor = (JavascriptExecutor) browser.driver;
         jsExecutor.executeScript("document.getElementsByClassName('popover')[0].parentNode.removeChild(document.getElementsByClassName('popover')[0])");
     }
+    
+    public void hoverClickAndViewGiverPhotoOnTableCell(int questionBodyIndex, int tableRow, String urlRegex) throws Exception {
+        String idOfQuestionBody = "questionBody-" + questionBodyIndex;
+        WebElement photoDiv = browser.driver.findElement(By.id(idOfQuestionBody))
+                                            .findElements(By.cssSelector(".dataTable tbody tr"))
+                                            .get(tableRow)
+                                            .findElements(By.cssSelector("td"))
+                                            .get(0)
+                                            .findElement(By.className("profile-pic-icon-hover"));
+        Actions actions = new Actions(browser.driver);
+        actions.moveToElement(photoDiv).build().perform();      
+        waitForElementToAppear(By.cssSelector(".popover-content"));
+        
+        JavascriptExecutor jsExecutor = (JavascriptExecutor) browser.driver;
+        jsExecutor.executeScript("document.getElementsByClassName('popover-content')[0]"
+                + ".getElementsByTagName('a')[0].click();");
+        
+        waitForElementToAppear(By.cssSelector(".popover-content > img"));
+        
+        AssertHelper.assertContainsRegex(urlRegex, 
+                browser.driver.findElements(By.cssSelector(".popover-content > img"))
+                              .get(0)
+                              .getAttribute("src"));
+        
+        jsExecutor.executeScript("document.getElementsByClassName('popover')[0].parentNode.removeChild(document.getElementsByClassName('popover')[0])");
+    }
+    
+    public void hoverClickAndViewRecipientPhotoOnTableCell(int questionBodyIndex, int tableRow, String urlRegex) throws Exception {
```

Comment:
any specific reason we have a separate function for this? I would suggest creating a generic function that takes in the column number as a parameter, then both these functions can call that one with the appropriate column number